presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoConnection.java

```diff
@@ -106,7 +108,7 @@
         this.catalog.set(uri.getCatalog());
         this.user = uri.getUser();
         this.applicationNamePrefix = uri.getApplicationNamePrefix();
-
+        this.extraCredentials = EXTRA_CREDENTIALS.getValue(uri.getProperties()).orElse(ImmutableMap.of());
```

Comment:
Move this to a getter `getExtraCredentials()` on `PrestoDriverUri`. This keeps the `ConnectionProperty` stuff encapsulated inside that class.